samplers/aws/xray/internal/rule.go

```diff
@@ -0,0 +1,213 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "go.opentelemetry.io/contrib/samplers/aws/xray/internal"
+
+import (
+	"sync/atomic"
+	"time"
+
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Rule represents a sampling rule which contains rule properties and reservoir which keeps tracks of sampling statistics of a rule
+type Rule struct {
+	samplingStatistics samplingStatistics
```

Comment:
Believe this needs to be a pointer

Reply:
Nah it does not necessarily have to be a pointer. I have tested the code and works as expected with this non-pointer declaration.

Reply:
The deep copy will copy the struct so I can't see how the statistics could be updated during refresh if this isn't a pointer. If you think it is, please explain the flow

Reply:
yep I changed it to pointer. If we don't change it to pointer then values get updated when we assign deep copy back to centralized object but making this struct pointer makes more sense since values would get updated even when deep copy logic gets executed. So changed it to make this a pointer.